lending/loan_management/doctype/loan_repayment_schedule/loan_repayment_schedule.py

```diff
@@ -941,22 +973,29 @@ def add_broken_period_interest(
 	):
 		interest_amount = flt(balance_amount * flt(rate_of_interest) * additional_days / (365 * 100))
 
-		if schedule_field == "repayment_schedule":
-			self.broken_period_interest += interest_amount
-
-		payment_date = add_months(payment_date, -1)
-		self.add_repayment_schedule_row(
-			payment_date,
-			0,
-			interest_amount,
-			interest_amount,
-			balance_amount,
-			additional_days,
-			repayment_schedule_field=schedule_field,
-			principal_share_percentage=principal_share_percentage,
-			interest_share_percentage=interest_share_percentage,
+		bpi_recovery_method = frappe.db.get_value(
+			"Loan Product", self.loan_product, "bpi_recovery_method"
 		)
 
+		if bpi_recovery_method == "Upfront Deduction":
+			if schedule_field == "repayment_schedule":
+				self.broken_period_interest += interest_amount
```

Comment:
This should be updated for all methods, not just "Upfront Deduction"
The value may be fetched in reports irrespective of the method